test-node.bash

```diff
@@ -253,51 +300,56 @@ fi
 if $blockscout; then
     NODES="$NODES blockscout"
 fi
-if $force_build; then
-  echo == Building..
-  if $dev_build_nitro; then
-    if ! [ -n "${NITRO_SRC+set}" ]; then
-        NITRO_SRC=`dirname $PWD`
-    fi
-    if ! grep ^FROM "${NITRO_SRC}/Dockerfile" | grep nitro-node 2>&1 > /dev/null; then
-        echo nitro source not found in "$NITRO_SRC"
-        echo execute from a sub-directory of nitro or use NITRO_SRC environment variable
-        exit 1
-    fi
-    docker build "$NITRO_SRC" -t nitro-node-dev --target nitro-node-dev
+
+
+if $dev_nitro && $build_dev_nitro; then
+  echo == Building Nitro
+  if ! [ -n "${NITRO_SRC+set}" ]; then
+      NITRO_SRC=`dirname $PWD`
   fi
-  if $dev_build_blockscout; then
-    if $blockscout; then
-      docker build blockscout -t blockscout -f blockscout/docker/Dockerfile
-    fi
+  if ! grep ^FROM "${NITRO_SRC}/Dockerfile" | grep nitro-node 2>&1 > /dev/null; then
+      echo nitro source not found in "$NITRO_SRC"
+      echo execute from a sub-directory of nitro or use NITRO_SRC environment variable
+      exit 1
   fi
+  docker build "$NITRO_SRC" -t nitro-node-dev --target nitro-node-dev
+fi
+if $dev_blockscout && $build_dev_blockscout; then
+  if $blockscout; then
+    echo == Building Blockscout
+    docker build blockscout -t blockscout -f blockscout/docker/Dockerfile
+  fi
+fi
 
+if $build_utils; then
   LOCAL_BUILD_NODES="scripts rollupcreator"
   if $tokenbridge || $l3_token_bridge; then
     LOCAL_BUILD_NODES="$LOCAL_BUILD_NODES tokenbridge"
   fi
-  docker compose build --no-rm $LOCAL_BUILD_NODES
+  UTILS_NOCACHE=""
+  if $force_build_utils; then
+      UTILS_NOCACHE="--no-cache"
+  fi
+  docker compose build --no-rm $UTILS_NOCACHE $LOCAL_BUILD_NODES
 fi
 
-if $dev_build_nitro; then
+if $dev_nitro; then
   docker tag nitro-node-dev:latest nitro-node-dev-testnode
 else
   docker pull $NITRO_NODE_VERSION
   docker tag $NITRO_NODE_VERSION nitro-node-dev-testnode
 fi
 
-if $dev_build_blockscout; then
-  if $blockscout; then
+if $blockscout; then
+  if $dev_blockscout; then
     docker tag blockscout:latest blockscout-testnode
-  fi
-else
-  if $blockscout; then
+  else
     docker pull $BLOCKSCOUT_VERSION
     docker tag $BLOCKSCOUT_VERSION blockscout-testnode
   fi
 fi
 
-if $force_build; then
+if $build_node_images; then
```

Comment:
I'm not sure if we need this as a separate step?
isn't everything this can do covered by previous builds?

Reply:
I think this is the only place we're building $NODES that I could find in the code.